### DISCUSSION TITLE
How to confirm Data is being sent out DataPin?

### DISCUSSION THREAD

**Discussion by italytower** [forkineye/ESPixelStick#795] [Q&A]
I am having issues where i cannot get the lights to change in test mode v3.2 or at all when I use the controller. I tested with another controller and the lights work just fine. I upgraded to the v4.5 and I see there is no more test mode from the controller to test display. I am stuck with determining how / why the lights are stuck on blue even after re-flashing. 

I have a v3 pixel stick - now running 4.5

**Comment by MartinMueller2003** [comment]
Test mode in V4 is not gone. It has been move to input 2 and named "Effects"

**Comment by MartinMueller2003** [comment]
V4 always sends out data. Some pixels come up with a fixed color on boot and stay that way until they get data. It sounds like you do not have the Data plus V- wires properly connected to the pixels.

**Reply by italytower** [reply]
Thanks for the response. I've had it configured this way for couple years now with no issues. It most deff connected to the v+ d+ and v- for the output. I tried to bypass the connecter and hold the D+ on the solder joint itself no luck. Still static blue lights. 

If it was a bad v- output what would you recommend to resolve it? 

I was able to flash it to any version I tried then all ;) no luck.

**Reply by italytower** [reply]
![image](https://github.com/user-attachments/assets/c73ff86d-fae6-4da4-aaef-cf84632452fe)

**Reply by italytower** [reply]
> V4 always sends out data. Some pixels come up with a fixed color on boot and stay that way until they get data. It sounds like you do not have the Data plus V- wires properly connected to the pixels.

Would you happen to be able to share more details based on what I posted in response

**Comment by MartinMueller2003** [comment]
It really sounds like you may have a bad driver / level shifter. Use a voltmeter to measure the D+ line. It should be between 2v and 4v.
